wolfssl/wolfcrypt/port/Espressif/esp32-crypt.h

```diff
@@ -22,13 +22,143 @@
 
 #define __ESP32_CRYPT_H__
 
-#include "wolfssl/wolfcrypt/settings.h"
-#include <wolfssl/wolfcrypt/types.h> /* for MATH_INT_T */
-
-#include "esp_idf_version.h"
-#include "esp_types.h"
-#include "esp_log.h"
-
+#include "sdkconfig.h" /* ensure ESP-IDF settings are available everywhere */
+
+/* wolfSSL  */
+#include <wolfssl/wolfcrypt/settings.h> /* references user_settings.h */
+#include <wolfssl/wolfcrypt/types.h>    /* for MATH_INT_T */
+
+/* Espressif */
+#include <esp_idf_version.h>
+#include <esp_types.h>
+#include <esp_log.h>
+
+/* exit codes to be used in tfm.c, sp_int.c, integer.c, etc. */
+/* TODO what numbers do we really want to use? */
+#define MP_HW_ERROR    (-106) /* hardware error, consider SW fallback  */
+#define MP_HW_BUSY     (-107) /* assigned -7 to match SP_HW_BUSY       */
```

Comment:
is SP_HW_BUSY an esp32 value? grep'ing for it in wolfSSL didn't show any hits

Reply:
The short answer is I could just remove the comment... :)   however... 

That reminds me of https://github.com/wolfSSL/wolfssl/pull/6565; @dgarske has a valid point that I was proposing to add codes not actually (yet) used. My intention is that they would have at least been consistent between [integer.h, sp_int.h, and tfm.h](https://github.com/wolfSSL/wolfssl/pull/6565/files). The plan is to add HW acceleration in `integer.h` and/or `sp_int.h` libraries. (Today ESP HW accleration is only available in the `tfm.h` code).

So in the meantime I added the error codes to the [esp32-crypt.h](https://github.com/gojimmypi/wolfssl/blob/c18fb50744dbf35bbc49fa5ffe0706182a691334/wolfssl/wolfcrypt/port/Espressif/esp32-crypt.h#L36). See [my comments in #6565](https://github.com/wolfSSL/wolfssl/pull/6565#issuecomment-1629903023).

What do you think would be best for these error codes?

Reply:
marking resolved, think this is close to a duplicate of the other WC_HW_WAIT_E comment and that it will be resolved at the same time as the other